test/providers/cuda_helpers.cpp

```diff
@@ -0,0 +1,288 @@
+        return -1;
+    }
+    *(void **)&libcu_ops.cuMemAllocHost =
+        util_get_symbol_addr(cuDlHandle.get(), "cuMemAllocHost_v2", lib_name);
+    if (libcu_ops.cuMemAllocHost == nullptr) {
+        fprintf(stderr, "cuMemAllocHost_v2 symbol not found in %s\n", lib_name);
+        return -1;
+    }
+    *(void **)&libcu_ops.cuMemAllocManaged =
+        util_get_symbol_addr(cuDlHandle.get(), "cuMemAllocManaged", lib_name);
+    if (libcu_ops.cuMemAllocManaged == nullptr) {
+        fprintf(stderr, "cuMemAllocManaged symbol not found in %s\n", lib_name);
+        return -1;
+    }
+    *(void **)&libcu_ops.cuMemFreeHost =
+        util_get_symbol_addr(cuDlHandle.get(), "cuMemFreeHost", lib_name);
+    if (libcu_ops.cuMemFreeHost == nullptr) {
+        fprintf(stderr, "cuMemFreeHost symbol not found in %s\n", lib_name);
+        return -1;
+    }
+    *(void **)&libcu_ops.cuMemsetD32 =
+        util_get_symbol_addr(cuDlHandle.get(), "cuMemsetD32_v2", lib_name);
+    if (libcu_ops.cuMemsetD32 == nullptr) {
+        fprintf(stderr, "cuMemsetD32_v2 symbol not found in %s\n", lib_name);
+        return -1;
+    }
+    *(void **)&libcu_ops.cuMemcpyDtoH =
+        util_get_symbol_addr(cuDlHandle.get(), "cuMemcpyDtoH_v2", lib_name);
+    if (libcu_ops.cuMemcpyDtoH == nullptr) {
+        fprintf(stderr, "cuMemcpyDtoH_v2 symbol not found in %s\n", lib_name);
+        return -1;
+    }
+    *(void **)&libcu_ops.cuPointerGetAttributes = util_get_symbol_addr(
+        cuDlHandle.get(), "cuPointerGetAttributes", lib_name);
+    if (libcu_ops.cuPointerGetAttributes == nullptr) {
+        fprintf(stderr, "cuPointerGetAttributes symbol not found in %s\n",
+                lib_name);
+        return -1;
+    }
+
+    return 0;
+}
+
+#else  // USE_DLOPEN
+int InitCUDAOps() {
+    // CUDA is linked statically but we prepare ops structure to
+    // make test code consistent
+    libcu_ops.cuInit = cuInit;
+    libcu_ops.cuCtxCreate = cuCtxCreate;
+    libcu_ops.cuCtxDestroy = cuCtxDestroy;
+    libcu_ops.cuDeviceGet = cuDeviceGet;
+    libcu_ops.cuMemAlloc = cuMemAlloc;
+    libcu_ops.cuMemAllocHost = cuMemAllocHost;
+    libcu_ops.cuMemAllocManaged = cuMemAllocManaged;
+    libcu_ops.cuMemFree = cuMemFree;
+    libcu_ops.cuMemFreeHost = cuMemFreeHost;
+    libcu_ops.cuMemsetD32 = cuMemsetD32;
+    libcu_ops.cuMemcpyDtoH = cuMemcpyDtoH;
+    libcu_ops.cuPointerGetAttributes = cuPointerGetAttributes;
+
+    return 0;
+}
+#endif // USE_DLOPEN
+
+static int init_cuda_lib(void) {
+    CUresult result = libcu_ops.cuInit(0);
+    if (result != CUDA_SUCCESS) {
+        return -1;
+    }
+    return 0;
+}
+
+int cuda_fill(CUcontext context, CUdevice device, void *ptr, size_t size,
+              const void *pattern, size_t pattern_size) {
+
+    (void)context;
+    (void)device;
+    (void)pattern_size;
+
+    // TODO support patterns > sizeof(unsigned int)
+    assert(pattern_size <= sizeof(unsigned int));
```

Comment:
What will happen here in the Release version if ` pattern_size > sizeof(unsigned int)`?

Reply:
done